File: Assets/Scripts/UI/Messages/SystemMessage.cs
public enum MessageType {
    USER_LOGGED_IN,
    USER_LOGGED_OFF
}

public abstract class SystemMessage {
    private byte[] _data;

    public SystemMessage() {}
    public SystemMessage(byte[] data) {
        _data = data;
    }
    
    public abstract override string ToString(); 
}

File: java-server/src/main/java/com/shnok/serverpackets/SystemMessage.java
package com.shnok.serverpackets;

public class SystemMessage extends ServerPacket {

    public MessageType _type;

    public SystemMessage(MessageType type) {
        super((byte) 0x03);
        _type = type;
    }

    public SystemMessage(MessageType type, String value) {
        super((byte) 0x03);
        _type = type;

        writeB((byte) type.ordinal());
        writeS(value);

        buildPacket();
    }

    public enum MessageType {
        USER_LOGGED_IN,
        USER_LOGGED_OFF
    }
}
